common/persistence/class.PhpRedisDriver.php

```diff
@@ -14,72 +14,127 @@
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  *
- * Copyright (c) 2013 (original work) Open Assessment Technologies SA (under the project TAO-PRODUCT);
+ * Copyright (c) 2013-2017 (original work) Open Assessment Technologies SA (under the project TAO-PRODUCT);
  *
  * @author Lionel Lecaque  <lionel@taotesting.com>
  * @license GPLv2
- * @package 
- 
+ * @package
  *
  */
 class common_persistence_PhpRedisDriver implements common_persistence_AdvKvDriver
 {
 
-    const DEFAULT_PORT = 6379;
+    const DEFAULT_PORT     = 6379;
+    const DEFAULT_ATTEMPT  = 3;
+    const DEFAULT_TIMEOUT  = 5;
+    const RETRY_DELAY      = 500000; // Eq to 0.5s
 
     /**
      * @var Redis
      */
     private $connection;
 
     /**
+     * @var $params
+     */
+    private $params;
+
+    /**
      * (non-PHPdoc)
      * @see common_persistence_Driver::connect()
      */
     function connect($key, array $params)
     {
+        $this->params = $params;
+        $this->connectionSet($params);
+
+        return new common_persistence_AdvKeyValuePersistence($params, $this);
+    }
+
+    function connectionSet(array $params)
```

Comment:
pls add phpdoc block (not sure on difference between connect() and connectionSet())